newrelic_python_agent/plugins/mysql.py

```diff
@@ -0,0 +1,922 @@
+            if total > 0:
+                pct_query_cache_memory_in_use = 100.0 - ((free / total) * 100.0)
+
+            self.update_metric("newrelic/pct_query_cache_memory_in_use", pct_query_cache_memory_in_use)
+
+        # Temp Table
+        vals = self.get_values(["status/created_tmp_tables", "status/created_tmp_disk_tables"])
+        if vals:
+            tmp_tables, tmp_tables_disk = vals
+
+            pct_tmp_tables_written_to_disk = 0.0
+            if tmp_tables > 0:
+                pct_tmp_tables_written_to_disk = (tmp_tables_disk / tmp_tables) * 100.0
+
+            self.update_metric("newrelic/pct_tmp_tables_written_to_disk", pct_tmp_tables_written_to_disk)
+
+    def derive_newrelic_slaves(self):
+        """
+        Derive newrelic status metrics about slaves
+        """
+        if self.has_slave_data is True:
+            self.update_metric("newrelic/replication_lag", self.sum_of(["slave/seconds_behind_master"]))
+
+            # both need to be YES, which is 1
+            running = self.sum_of(["slave/slave_io_running", "slave/slave_sql_running"])
+            if running is not None:
+                replication_status = 1.0
+                if running == 2:
+                    replication_status = 0.0
+                self.update_metric("newrelic/replication_status", replication_status)
+            self.update_metric("newrelic/slave_relay_log_bytes", self.sum_of(["slave/relay_log_pos"]))
+            self.update_metric("newrelic/master_log_lag_bytes", self.diff_of(["slave/read_master_log_pos",
+                                                                             "slave/exec_master_log_pos"]))
+        else:  # This is a hack because the NR UI can't handle it missing for graphs
+            self.update_metric("newrelic/replication_lag", 0.0)
+            self.update_metric("newrelic/replication_status", 0.0)
+            self.update_metric("newrelic/slave_relay_log_bytes", 0.0)
+            self.update_metric("newrelic/master_log_lag_bytes", 0.0)
+
+    def parse_metric_value(self, value):
+        """
+        Parse the values from mysql, converting them to floats when necessary
+            on|yes|true => 1
+            off|no|false => 0
+            null => -1
+
+        :param str value: The value from mysql
+        :return: the parsed value
+        :rtype: float or None
+        """
+        if isinstance(value, str):
+            if value == "":
+                return None
+
+            # yes|true|on
+            if self.is_true.match(value):
+                return 1
+            # no|false|off
+            if self.is_false.match(value):
+                return 0
+            if self.is_null.match(value):
+                return -1
+
+            # anything else, try to convert it to a float
+            try:
+                r = float(value)
+                return r
+            except:
+                pass
+
+            return None
+
+        return value
+
+    def is_number(self, value):
```

Comment:
Ideally (no need for now) this would be moved to the base plugin at some point. With Python 3 support as well (long has been removed in Python 3).

Reply:
agreed! i don't have much experience with python 3, so you guys will have to keep me honest in that respect :)